scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -309,284 +310,453 @@ Note that while we applied this technique to assign cell types using the ADT dat
+Otherwise, the components are very similar: each row is still a gene, for example, and additional data about the samples are stored in the `colData`.
+In fact, the `SingleCellExperiment` object is derived from a `SummarizedExperiment`, with some extra slots that are more relevant to single-cell data.
 
 What information do we have for the samples?
 
-```{r}
-colData(blueprint_encode)
+```{r explore reference sample data}
+colData(monaco_ref)
 ```
 
-Let's see how many samples of each type we have:
+There are three main columns for the sample data:
+
+- `label.main` is a more general cell type assignment.  
+
+- `label.fine` is a fine-level cell type with more specific labels.
+The exact level of granularity of these `main` and `fine` designations (and indeed the label names themselves) will vary among datasets, so it is important to look at the reference to see whether it is suitable for your application.
 
-```{r}
-data.frame(colData(blueprint_encode)) %>% 
+- `label.ont` is a standardized [Cell Ontology](https://www.ebi.ac.uk/ols/ontologies/cl) identifier. 
+Using the cell ontology can allow for more complex representations of the relationships among different cell types, but investigating that is beyond the scope of this workshop. 
+
+Another component we would like to explore is how many of each of these cell types we have in the dataset. 
+A bit of quick `dplyr` wrangling can give us the answer.
+
+```{r count cell types}
+colData(monaco_ref) |> 
+  as.data.frame() |>
   dplyr::count(label.main, label.fine)
 ```
 
+This is pretty good! 
+Most cell types have 4 replicates, which is more replicates than we often find.
+
 ### What does `SingleR` do?
 
-`SingleR` builds a model from a set of training data, and then uses that model to classify cells (or groups of cells) in new data sets.
-We will show how to separate these results
+As mentioned earlier, `SingleR` builds a model from a set of training data, and then uses that model to classify cells (or groups of cells) in new datasets.
 
-This time we will do the training and classification in a single step, using the convenience function `SingleR::SingleR()`.
+`SingleR` works by first identifying a set of marker genes that can be used to differentiate among the cell types in the reference dataset. 
+It does this by performing pairwise comparisons among all of the cell types, and retaining the top set of genes differentiating each pair.
+The idea is that this set of genes will be the most informative for differentiating cell types.
 
-```{r}
+Then, for each cell, `SingleR` calculates the Spearman correlation between expression of that cell and each cell type (using the only the genes chosen earlier).
+Notably, this is a non-parametric correlation, so the scaling and normalization that we apply (or don't) should not matter!
+(But if you have full-length transcripts from another technology, you will probably want to convert to Transcripts per Million (TPM).)
```

Comment:
```suggestion
[But if you have full-length transcripts from another technology, you will probably want to convert to Transcripts per Million (TPM).]
```

Reply:
Is this because there are nested parentheses? If it really bothers you, I would go the other way, with square internal parens. But I'll just unnest:
```suggestion
Note that if you used a single-cell technology that produces full-length transcripts (i.e., SMART-seq), you will probably want to convert your counts to Transcripts per Million (TPM), to allow more consistent ranking among transcripts of different lengths. 
```